### ISSUE TITLE
[Dashboards]: Search API is not consistent while loading dashboard list in a folder

### ISSUE THREAD

**Issue by aishyandapalli** [grafana/grafana#75586]
### What happened?

We have few folders in Grafana. When we browse the folder through Grafana UX and load the dashboards, the list seems to be consistent but, when we load it through URL, there are too many dashboards in the folder.

I went through the network calls and found that search API is querying for `folderIDs=0` instead of the actual `folderID` which is returning all dashboards.

It's only happening for few folders and not all.

### What did you expect to happen?

Search API should query for the actual folderID instead of 0

### Did this work before?

Not sure

### How do we reproduce it?

1. We chose a particular folder. Browsed it through Grafana UX and opened dashboard list in the folder
2. Refresh the URL
3. We will see too many dashboards not related to the folder

### Environment (with versions)?

Grafana: 9.2.8
OS: Linux
Browser: Chrome

### Grafana platform?

None

**Comment by tonypowa** [comment]
hi @aishyandapalli 

thank you for opening this issue

I could not repro the same behaviour from my end.  (v.9.2.8)

Just to double check that I am following the same steps, when you open the folder, your browser url looks like:

`/dasboards/f/KEdaz46vb/folder-name`

and then in the network inspector you filter by `folderid`

![image](https://github.com/grafana/grafana/assets/45235678/1170f2cd-d2f3-4ae4-b4bc-5cfddb2833c9)

I just see one `folderid` match (7)

**Comment by tarlitongodoy** [comment]
Hi @tonypowa ,

We are also having this problem, we can reproduce it with the following steps:

- add lots of folders

```python
import requests
session = requests.session()
grafana_token = '<token>'
for i in range(2000):
    resp = requests.post('http://localhost:3000/api/folders', json={"title":f"folder_{i}"}, headers={'Authorization': f'Bearer {grafana_token}'})
    print(resp.json())
```

- add a folder that will alphabetically come after the ones from previous step, for example: `other_folder`
- make sure that on the `Browse` page the sorting is `Alphabetically (A-Z)`
- get the link to the last folder on the list, for example: http://localhost:3000/dashboards/f/Pg1S1UOIk/other-folder
- the response on the UI will be all folders
- check network tab from chrome, we'll see: http://localhost:3000/api/search?limit=1000&sort=alpha-asc&folderIds=0

Tested on version: 9.4.17, not sure if this happens on newer versions.

Thanks!

**Comment by tonypowa** [comment]
Thanks for the script and the clear repro steps.
I have reproduced it in v.9.4.17 but it is not reproducible in a later version ( v.10.2.2 )

![folder](https://github.com/grafana/grafana/assets/45235678/dac4b675-1c01-43dd-a828-b462e4b7d08e)

Detail of the network tab:
![image](https://github.com/grafana/grafana/assets/45235678/1d412521-8904-4789-a7a0-7f78ce1e9168)

I will forward this issue to the Dashboard team so they are aware of it (even though the issue is not present from later versions)

**Comment by ashharrison90** [comment]
dupe of https://github.com/grafana/grafana/issues/64756 which was fixed and released in grafana 10.2.0 🥳
